src/test/resources/examples/singleAllOf/controllers/ktor/Controllers.kt

```diff
@@ -66,3 +71,29 @@ public interface TestController {
             BadRequestException("Header " + name + " is required")
     }
 }
+
+/**
+ * Decorator for Ktor's ApplicationCall that provides type safe variants of the [respond] functions.
+ *
+ * It can be used as a drop-in replacement for [io.ktor.server.application.ApplicationCall].
+ *
+ * @param R The type of the response body
+ */
+public class TypedApplicationCall<R : Any> private constructor(
+    private val applicationCall: ApplicationCall,
+) : ApplicationCall by applicationCall {
+    @Suppress("unused")
+    public suspend inline fun <reified T : R> respondTyped(message: T) {
+        respond(message)
+    }
```

Comment:
Why the `T` generic and not just `message: R`?

Reply:
I am not 100% why, but this setup is how I managed to make it work with the original call's `respond`:

<img width="540" alt="Screenshot 2024-04-28 at 13 42 20" src="https://github.com/cjbooms/fabrikt/assets/11440456/167ec80d-dfd1-4bf4-8395-a7c6462e1967">

But maybe I am missing something?

We can also use the `respond` function variants that take a TypeInfo (https://api.ktor.io/ktor-server/ktor-server-core/io.ktor.server.response/respond.html):

```kotlin
public class TypedApplicationCall<R : Any> (
  private val applicationCall: ApplicationCall,
) : ApplicationCall by applicationCall {
  @Suppress("unused")
  public suspend inline fun <reified R> respondTyped(message: R) {
    respond(message, typeInfo<R>())
  }

  @Suppress("unused")
  public suspend inline fun <reified R> respondTyped(status: HttpStatusCode, message: R) {
    respond(status, message, typeInfo<R>())
  }
}
```

Let me know what you think is best.

Reply:
I guess it doesn't really matter because either way, the generated `Response` class is final (or rather, not `open`), so the generic argument will always be the same type anyway, right?

Reply:
Yes, the models are final, so `R` will always be the model and not a subclass.